scio-core/src/main/scala/com/spotify/scio/bigquery/dynamic/package.scala

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2017 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.bigquery
+
+import com.google.api.services.bigquery.model.TableSchema
+import com.spotify.scio.bigquery.types.BigQueryType.HasAnnotation
+import com.spotify.scio.io.Tap
+import com.spotify.scio.testing.DynamicBigQueryIO
+import com.spotify.scio.util.Functions
+import com.spotify.scio.values.SCollection
+import org.apache.beam.sdk.io.gcp.{bigquery => bqio}
+import org.apache.beam.sdk.io.gcp.bigquery.BigQueryIO.Write.{CreateDisposition, WriteDisposition}
+import org.apache.beam.sdk.io.gcp.bigquery.{DynamicDestinations, TableDestination}
+import org.apache.beam.sdk.values.ValueInSingleWindow
+
+import scala.concurrent.Future
+import scala.reflect.ClassTag
+import scala.reflect.runtime.universe._
+
+package object dynamic {
+
+  implicit class DynamicBigQuerySCollection[T](val self: SCollection[T]) extends AnyVal {
+
+    /**
+      * Save this SCollection to multiple BigQuery tables using the table and schema
+      * specified by the [[org.apache.beam.sdk.io.gcp.bigquery.DynamicDestinations]] object.
+      * @group output
+      */
+    def saveAsBigQuery(name: String,
+                       destinations: DynamicDestinations[T, _],
+                       formatFn: T => TableRow,
+                       writeDisposition: WriteDisposition,
+                       createDisposition: CreateDisposition): Future[Tap[T]] = {
+      if (self.context.isTest) {
+        self.context.testOut(DynamicBigQueryIO(name))(self)
```

Comment:
Not sure if it makes sense to capture this in test, since we can't easily verify the dynamic destination logic.
Get rid of `DynamicBigQueryIO` and throw an error here?

Reply:
@nevillelyh removed